core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -5294,29 +5274,55 @@ public void setKeyRotationTime(@Nullable Date time) {
      * The given time cannot be in the future.
      * </p>
      */
-    public void setKeyRotationTime(long unixTimeSeconds) {
-        checkArgument(unixTimeSeconds <= TimeUtils.currentTimeSeconds(), "Given time (%s) cannot be in the future.",
-                TimeUtils.dateTimeFormat(unixTimeSeconds * 1000));
-        vKeyRotationTimestamp = unixTimeSeconds;
+    public void setKeyRotationTime(Instant keyRotationTime) {
+        checkArgument(keyRotationTime.compareTo(TimeUtils.currentTime()) <= 0,
+                "Given time (%s) cannot be in the future.",
+                TimeUtils.dateTimeFormat(keyRotationTime.toEpochMilli()));
+        vKeyRotationTime = Optional.of(keyRotationTime);
         saveNow();
     }
 
+    /** @deprecated use {@link #setKeyRotationTime(Instant)} */
+    @Deprecated
+    public void setKeyRotationTime(long timeSecs) {
+        if (timeSecs == 0)
+            vKeyRotationTime = Optional.empty();
+        else
+            setKeyRotationTime(Instant.ofEpochSecond(timeSecs));
+    }
+
+    /** @deprecated use {@link #setKeyRotationTime(Instant)} */
+    @Deprecated
+    public void setKeyRotationTime(@Nullable Date time) {
+        if (time == null)
+            vKeyRotationTime = Optional.empty();
+        else
+            setKeyRotationTime(Instant.ofEpochMilli(time.getTime()));
+    }
+
     /**
-     * Returns the key rotation time, or null if unconfigured. See {@link #setKeyRotationTime(Date)} for a description
+     * Returns the key rotation time, or null if unconfigured. See {@link #setKeyRotationTime(Instant)} for a description
      * of the field.
      */
+    public Optional<Instant> getKeyRotationTimeInstant() {
```

Comment:
I suppose it would also be possible to use `Instant.MAX` (aka "never") for the case where there is no rotation enabled.

Reply:
Currently (as a long) we use 0 as the magic constant for "never". I'm a bit surprised by this recommendation to use a magic value. I thought `Optional` is exactly meant for this, or `null`.